doc/contribute/coding_guidelines/index.rst

```diff
@@ -1255,6 +1255,39 @@ Related GitHub Issues and Pull Requests are tagged with the `Inclusive Language
 .. _CAN in Automation Inclusive Language news post: https://www.can-cia.org/news/archive/view/?tx_news_pi1%5Bnews%5D=699&tx_news_pi1%5Bday%5D=6&tx_news_pi1%5Bmonth%5D=12&tx_news_pi1%5Byear%5D=2020&cHash=784e79eb438141179386cf7c29ed9438
 .. _CAN in Automation Inclusive Language: https://can-newsletter.org/canopen/categories/
 
+
+Rule A.3: Macro name collisions
+===============================
+
+Severity
+--------
+
+Required
+
+Description
+-----------
+
+Commonly used macros such as  ``MIN``, ``MAX``, ``ARRAY_SIZE`` as well as other
+similar ones, must not be modified or protected to avoid name collisions with
+other implementations. In particular, they must not be prefixed to place them in
+a Zephyr-specific namespace, nor can they be conditionally excluded from
+compilation using ``#ifndef``.
+Instead, if a conflict arises with an existing definition originating from a
+:ref:`module <modules>`, the module's code itself needs to be modified (ideally
+upstream, alternatively via a change in Zephyr's own fork).
+
+Rationale
+---------
+
+Zephyr is an RTOS that comes with additional functionality and dependencies in
+the form of modules.  Those modules are typically independent projects that may
+use macro names that can conflict with other modules or with Zephyr itself.
+Since, in the context of this documentation, Zephyr is considered the central or
+main project, it is it that should implement the non-namespaced versions of the
+macros. Given that Zephyr uses a fork of the corresponding upstream for each
```

Comment:
typo

```suggestion
Since, in the context of this documentation, Zephyr is considered the central or
main project, it should implement the non-namespaced versions of the
macros. Given that Zephyr uses a fork of the corresponding upstream for each
```